File: examples/governance_static_go/domain/registered_stakeholder.go
package domain

import "time"

type RegisteredStakeholder struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	Name string `json:"name"`
	Email string `json:"email"`
	Role string `json:"role"`
	Team string `json:"team"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e RegisteredStakeholder) EventName() string { return "RegisteredStakeholder" }

func (e RegisteredStakeholder) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/identity_domain/lib/identity_domain/stakeholder/events/registered_stakeholder.rb
module IdentityDomain
  class Stakeholder
    module Events
      class RegisteredStakeholder
        attr_reader :aggregate_id, :name, :email, :role, :team, :status, :occurred_at

        def initialize(aggregate_id: nil, name: nil, email: nil, role: nil, team: nil, status: nil)
          @aggregate_id = aggregate_id
          @name = name
          @email = email
          @role = role
          @team = team
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
